Numbers/ViewController/GridViewController.swift

```diff
@@ -0,0 +1,296 @@
+    }
+    
+    func addLine(fromPoint start: CGPoint, toPoint end:CGPoint, color: CGColor) {
+        let line = CAShapeLayer()
+        let linePath = UIBezierPath()
+        linePath.move(to: start)
+        linePath.addLine(to: end)
+        line.path = linePath.cgPath
+        line.strokeColor = color
+        line.lineWidth = 1
+        line.lineJoin = CAShapeLayerLineJoin.round
+        self.view.layer.addSublayer(line)
+    }
+    
+    func setArray() {
+        boxList.append(box1)
+        boxList.append(box2)
+        boxList.append(box3)
+        boxList.append(box4)
+        boxList.append(box5)
+        boxList.append(box6)
+        boxList.append(box7)
+        boxList.append(box8)
+        boxList.append(box9)
+        boxList.append(box10)
+        boxList.append(box11)
+        boxList.append(box12)
+    }
+    
+    func setLevel() {
+      levelLabel.text = "Level " + String(level - 3)
+    }
+    
+    func preStart() {
+        for index in 0...2 {
+            var randomNum = Int.random(in: 0 ..< boxList.count)
+            if(boxOrder.count > 0) {
+                while(randomNum == boxOrder[boxOrder.count - 1]) {
+                    randomNum = Int.random(in: 0 ..< boxList.count)
+                }
+            }
+            boxOrder.append(randomNum)
+        }
+    }
+    
+    
+    func start() {
+        var randomNum = Int.random(in: 0 ..< boxList.count)
+            if(boxOrder.count > 0) {
+                while(randomNum == boxOrder[boxOrder.count - 1]) {
+                    randomNum = Int.random(in: 0 ..< boxList.count)
+                }
+            }
+            boxOrder.append(randomNum)
+        showCircles()
+    }
+    
+    
+    func showCircles(fromIndex index: Int = 0) {
+        self.view.isUserInteractionEnabled = false
+        incorrectLabel.isHidden = true
+        guard index < boxOrder.count else {
+            self.view.isUserInteractionEnabled = true
+            return
+        }
+        boxList[boxOrder[index]].text = "O"
+        DispatchQueue.main.asyncAfter(deadline: .now() + secs) {
+            self.boxList[self.boxOrder[index]].text = ""
+            
+        }
+        
+        Timer.scheduledTimer(withTimeInterval: secs, repeats: false) { _ in
+            self.showCircles(fromIndex: index+1)
+        }
+    }
+    
+
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        incorrectLabel.isHidden = true
+        let touch: UITouch = touches.first as! UITouch
+        if(tapCount < level) {
+            var userGuess = getCorrectBoxWithCorrdinates(touchPoint: touch.location(in: view))
+            boxList[userGuess].textColor = UIColor.red
+            boxList[userGuess].text = "O"
+            DispatchQueue.main.asyncAfter(deadline: .now() + secs) {
+                self.boxList[userGuess].text = ""
+                self.boxList[userGuess].textColor = UIColor.black
+            }
+            if(userGuess == boxOrder[tapCount]) {
```

Comment:
make this into a function for easier readibility: if (isCorrectGuess(userGuess))